config/crd/bases/infrastructure.cluster.x-k8s.io_metal3remediations.yaml

```diff
@@ -0,0 +1,99 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.5.0
+  creationTimestamp: null
+  name: metal3remediations.infrastructure.cluster.x-k8s.io
+spec:
+  group: infrastructure.cluster.x-k8s.io
```

Comment:
Note that you will need to add a cluster role so that MHC in CAPI can create such CRs.
see example here:
https://github.com/medik8s/poison-pill/pull/3

but please note this is aimed for OCP and poison pill. There are some more details on that slack conversation

Reply:
Now when controller implementation is done within the CAPM3 the CRD are also defined in it. For changing CRDs to metal3 group we should move the whole controller outside the CAPM3. Right?

Reply:
Implementation of Remediation Controller is now part of the CAPM3:  https://github.com/metal3-io/cluster-api-provider-metal3 

So the controller is similar to metal3machine controller etc. When generated it within the CAPM3 all the apis and CRD defined in it belongs to this groupversion: https://github.com/metal3-io/cluster-api-provider-metal3/blob/master/api/v1alpha4/groupversion_info.go

If we would like to use something like remediation.metal3.io IMO we should take the controller outside the CAPM3 and make it stand alone controller like ip-address-manager: https://github.com/metal3-io/ip-address-manager/blob/master/api/v1alpha1/groupversion_info.go

Reply:
Maybe taking controller outside is the good idea anyway. Any thoughts?

Reply:
yeah, that makes sense to me.
In [medik8s](https://github.com/medik8s) we also went this path of having a separate operator for remediation controllers.

Reply:
Since the implementation for this almost ready, I would vote for keeping it inside CAPM3 and then add a separate controller and take it out of CAPM3 in a future workitem, we can also modify the proposal to reflect this.

Reply:
Following up on discussion in the community meeting: I don't see a lot of value in creating an extra repo for this. If it's something that CAPI interacts with then it seems to me that conceptually it belongs in the CAPM3 repo unless there's some technical reason for splitting it (which I understand there's not).

That said, I'm not clear why CAPI folks are apparently recommending that this not be part of the same API group as the Metal3Machine &c. CRDs, so I don't have all the context here.

Reply:
Since there could be many external remediation providers, there was a concern that using the same api group would increase the chances for a collision of two different remediators using the same kind.
I believe the same concern is applicable for all the existing machines so it's not completely different.

I'm not sure if CAPI will object using the same api group, and I believe it should technically work.
more context here:
https://kubernetes.slack.com/archives/C8TSNPY4T/p1615886930004900

in medik8s, we used different api group per remediator, and used role aggregation to grant MHC permissions to interact with those external api groups.

I'm fine with both ways. whatever works for you.

Reply:
I agree with @kashifest and @zaneb I would rather go with the infrastructure specific remediation and keep the controller within CAPM3. It make more sense to have remediationRequest and remediationTemplate defined in the same API group as Machines we want to remediate.